File: modules/coupon-api-service/src/main/java/com/thanh0x/coursedeal/CouponApiServiceApplication.java
package com.thanh0x.coursedeal;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan(basePackages = "com.thanh0x.coursedeal")
public class CouponApiServiceApplication {

    public static void main(String[] args) {
        SpringApplication.run(CouponApiServiceApplication.class, args);
    }
}

File: modules/coupon-api-service/src/main/kotlin/com/thanh0x/coursedeal/CouponApiServiceApplication.kt
package com.thanh0x.coursedeal

import com.thanh0x.coursedeal.config.ApiProperties
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication

@SpringBootApplication
@EnableConfigurationProperties(ApiProperties::class)
class CouponApiServiceApplication

@Suppress("SpreadOperator")
fun main(args: Array<String>) {
    // runApplication's vararg signature requires spreading the incoming args array.
    runApplication<CouponApiServiceApplication>(*args)
}
